File: src/main/java/org/example/state/ToolContext.java
package org.example.state;

public class ToolContext {
    private Tool currentState;

    public ToolContext() {
        this.currentState = new SelectionTool();
    }

    public void setCurrentState(Tool newState) {
        this.currentState = newState;
    }

    public void mouseDown() {
        currentState.mouseDown();
    }

    public void mouseUp() {
        currentState.mouseDown();
    }


}

File: Kotlin/src/state/ToolContext.kt
package state

class ToolContext {
    private var currentState: Tool = SelectionTool()

    fun setCurrentState(newState: Tool) {
        currentState = newState
    }

    fun mouseDown() {
        currentState.mouseDown()
    }

    fun mouseUp() {
        currentState.mouseUp()
    }
}